File: leetcode/implementation/1260. Shift 2D Grid.cpp
class Solution {
public:
    vector<vector<int>> shiftGrid(vector<vector<int>>& grid, int k) {
        vector<vector<int>> shiftedGrid = grid;
        int m = grid.size();
        int n = grid[0].size();
        while(k--) {
            for(int i = 0; i < m; i++) {
                for(int j = 0; j < n; j++) {
                    if(j+1 < n) shiftedGrid[i][j+1] = grid[i][j];
                    if(i+1 < m) shiftedGrid[i+1][0] = grid[i][n-1];
                }
            }
            shiftedGrid[0][0] = grid[m-1][n-1];
            grid = shiftedGrid;
        }
        return shiftedGrid;
    }
};

File: leetcode/implementation/1260. Shift 2D Grid/1260. Shift 2D Grid.py
import copy
class Solution:
    def shiftGrid(self, grid: List[List[int]], k: int) -> List[List[int]]:
        row_len = len(grid)
        col_len = len(grid[0])
        shiftedGrid = [[0] * col_len for _ in range(row_len)]
        for _ in range(k):
            for i in range(row_len):
                for j in range(col_len):
                    shiftedGrid[i][j] = grid[i][(j - 1) % col_len]
                shiftedGrid[i][0] = grid[(i-1) % row_len][col_len-1]
            shiftedGrid[0][0] = grid[row_len-1][col_len-1]
            grid = copy.deepcopy(shiftedGrid)
        return grid
